src/lib/utils/options/declaration.ts

```diff
@@ -92,16 +92,17 @@ export class OptionDeclaration {
                 }
                 break;
             case ParameterType.Map:
-                if (this.map !== 'object') {
+                const map = this.map;
+                if (map !== 'object') {
                     const key = value ? (value + '').toLowerCase() : '';
-                    if (key in this.map) {
-                        value = this.map[key];
+
+                    if (map instanceof Map && map.has(key)) {
+                        value = map.get(key);
+                    } else if (key in map) {
+                        value = map[key];
                     } else if (errorCallback) {
-                        if (this.mapError) {
-                            errorCallback(this.mapError);
-                        } else {
-                            errorCallback('Invalid value for option "%s".', this.name);
-                        }
+                        const message = this.mapError ? [this.mapError] : ['Invalid value for option "%s".', this.name];
+                        errorCallback(... message);
```

Comment:
Can we keep this code the same as it was before, I don't think there's any need to be clever here.